x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/RoleDescriptorStore.java

```diff
@@ -169,6 +168,25 @@ private void resolveRoleNames(Set<String> roleNames, ActionListener<RolesRetriev
         }, listener::onFailure));
     }
 
+    private Set<RoleDescriptor> skipRolesUsingDocumentOrFieldLevelSecurity(Set<RoleDescriptor> roleDescriptors) {
+        final Map<Boolean, Set<RoleDescriptor>> roles = roleDescriptors.stream()
+            .collect(Collectors.partitioningBy(RoleDescriptor::isUsingDocumentOrFieldLevelSecurity, Collectors.toSet()));
+
+        final Set<RoleDescriptor> rolesToSkip = roles.get(true);
+        logger.warn(
+            "User roles [{}] are disabled since they require document or field level security to determine user access. "
+                + "These security features [{}, {}] are not available under the current license. "
+                + "Access to documents or fields granted by above roles will be denied. "
+                + "To re-enable the roles, upgrade license to [{}] or above, or renew if it's expired.",
```

Comment:
I checked other messages related to licensing issues and we can probably simplify this message to:

> User roles [{}] are disabled because they require field or document level security. The current license is non-compliant for [field and document level security].

for a few reasons:
1. A bit more concise
2. `Access to documents or fields granted by above roles will be denied`. This is not entirely correct because the same access can be granted by other enabled roles. But it's hard to explain the details. So it's better to just leave it out and only state the direct impact (instead of the possible implications). 
3. License expiry has its own warning messages (see `LicenseService`). Downgraded license also has its dedicated messages. So I think we don't need suggest upgrade or renew license here.

Reply:
Sounds good!